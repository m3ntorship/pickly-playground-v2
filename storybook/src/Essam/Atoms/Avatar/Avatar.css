/*************   */

/* Avatars */

/*************   */
.avatar {
  @apply rounded-full overflow-hidden;
}

.avatar-img {
  @apply w-full h-full block object-cover;
}

/* Large */
.avatar.large {
  @apply w-14  h-14;
}

/* Large */

/* Medium */
.avatar.medium {
  @apply w-10  h-10;
}

/* Medium */

/* Medium */
.avatar.small {
  @apply w-8  h-8;
}

/* Medium */

/*************   */

/* Avatars */

/*************   */
