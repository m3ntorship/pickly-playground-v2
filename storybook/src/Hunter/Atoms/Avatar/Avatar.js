import React from "react";
import PropTypes from "prop-types";
import Person from "../../../img/large-avatar.png";
import Anon from "../../../img/large-anan.png";
import NotFilled from "../../../img/large-notfilled.png";

const Avatar = ({ size, type }) => {
  const ImageSize =
    size === "large"
      ? "w-14 h-14"
      : size === "medium"
      ? "w-10 h-10"
      : "w-8 h-8";

  const ImageSrc =
    type === "person" ? Person : type === "anon" ? Anon : NotFilled;

  return (
    <img className={`ml-4 mt-4 rounded-full ${ImageSize}`} src={ImageSrc} />
  );
};

Avatar.PropTypes = {
  size: PropTypes.oneOf(["large", "medium", "small"]),
  type: PropTypes.oneOf(["person", "anon", "notfilled"]),
};

export default Avatar;
