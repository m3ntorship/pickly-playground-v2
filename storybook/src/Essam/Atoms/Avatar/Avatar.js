import React from "react";
import PropTypes from "prop-types";
import className from "classnames";
import avaAnon from "../../../img/large-anan.png";
import avaNotFilled from "../../../img/large-notfilled.png";
import "./Avatar.css";

export const Avatar = ({ avaSize, avaType, avaUrl, ...props }) => {
  const avatarClasses = className("avatar", {
    large: avaSize === "large",
    medium: avaSize === "medium",
    small: avaSize === "small",
  });
  const avatarImage = className({
    [avaAnon]: avaType === "anonymous",
    [avaNotFilled]:
      avaType === "not-filled" || (!avaUrl && avaType === "filled"),
    [avaUrl]: avaUrl && avaType === "filled",
  });
  return (
    <figure className={avatarClasses}>
      <img src={avatarImage} alt="avatar" className="avatar-img" {...props} />
    </figure>
  );
};
Avatar.propTypes = {
  avaSize: PropTypes.oneOf(["small", "medium", "large"]).isRequired,
  avaType: PropTypes.oneOf(["anonymous", "not-filled", "filled"]).isRequired,
  avaUrl: PropTypes.string,
};

Avatar.defaultProps = {
  avaSize: "large",
  avaType: "not-filled",
};
