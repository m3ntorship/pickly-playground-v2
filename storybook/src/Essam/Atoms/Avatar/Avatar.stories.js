import React from "react";
import { Avatar } from "./Avatar";
import avaImg from "../../../img/large-avatar.png";

export default {
  title: "Essam/Atoms/Avatars",
  component: Avatar,
};

const Template = (args) => <Avatar {...args}></Avatar>;

export const DefaultAvatar = Template.bind({});
DefaultAvatar.args = {
  avaSize: "large",
  avaType: "filled",
  avaUrl: avaImg,
};
