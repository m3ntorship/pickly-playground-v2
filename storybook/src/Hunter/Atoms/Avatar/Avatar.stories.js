import React from "react";
import Avatar from "./Avatar";

export default {
  title: "Hunter/Atom/Avatar",
  component: Avatar,
  argTypes: {
    size: {
      control: {
        type: "radio",
        options: ["large", "medium", "small"],
      },
    },
    type: {
      control: {
        type: "radio",
        options: ["person", "anon", "notfilled"],
      },
    },
  },
};

const Template = (args) => <Avatar {...args} />;

export const avatar = Template.bind({});
avatar.args = {
  size: "large",
  type: "person",
};
