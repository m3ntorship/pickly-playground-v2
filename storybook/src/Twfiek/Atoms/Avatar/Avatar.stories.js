import React from "react";
import Avatar from "./Avatar";

const defaultExport = {
  title: "Twfiek/Atoms/Avatar",
  component: Avatar,
};

export default defaultExport;

const Template = (args) => <Avatar {...args} />;

export const Default = Template.bind({});
Default.args = {
  variant: "filled",
  size: "lg",
};
