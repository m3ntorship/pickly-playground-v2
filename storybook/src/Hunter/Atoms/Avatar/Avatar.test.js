import React from "react";
import renderer from "react-test-renderer";
import Avatar from "./Avatar";

describe("Avatar Component", () => {
  it("should render (person-large)  when we apply nothing ", () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (person) when we apply (person) ", () => {
    const tree = renderer.create(<Avatar type="person" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (anon) when we apply (anon)", () => {
    const tree = renderer.create(<Avatar type="anon" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (notfilled)  when we apply (notfilled) ", () => {
    const tree = renderer.create(<Avatar type="notfilled" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (meduim) when we apply (meduim) ", () => {
    const tree = renderer.create(<Avatar size="meduim" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (small) when we apply (small) ", () => {
    const tree = renderer.create(<Avatar size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render (large) when we apply (large) ", () => {
    const tree = renderer.create(<Avatar size="large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
