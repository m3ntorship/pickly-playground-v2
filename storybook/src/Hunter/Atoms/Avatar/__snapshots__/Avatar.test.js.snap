// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component should render (anon) when we apply (anon) 1`] = `
<img
  className="ml-4 mt-4 rounded-full w-8 h-8"
  src="large-anan.png"
/>
`;

exports[`Avatar Component should render (large) when we apply (large)  1`] = `
<img
  className="ml-4 mt-4 rounded-full w-14 h-14"
  src="large-notfilled.png"
/>
`;

exports[`Avatar Component should render (meduim) when we apply (meduim)  1`] = `
<img
  className="ml-4 mt-4 rounded-full w-8 h-8"
  src="large-notfilled.png"
/>
`;

exports[`Avatar Component should render (notfilled)  when we apply (notfilled)  1`] = `
<img
  className="ml-4 mt-4 rounded-full w-8 h-8"
  src="large-notfilled.png"
/>
`;

exports[`Avatar Component should render (person) when we apply (person)  1`] = `
<img
  className="ml-4 mt-4 rounded-full w-8 h-8"
  src="large-avatar.png"
/>
`;

exports[`Avatar Component should render (person-large)  when we apply nothing  1`] = `
<img
  className="ml-4 mt-4 rounded-full w-8 h-8"
  src="large-notfilled.png"
/>
`;

exports[`Avatar Component should render (small) when we apply (small)  1`] = `
<img
  className="ml-4 mt-4 rounded-full w-8 h-8"
  src="large-notfilled.png"
/>
`;
